pkg/controller/validatingadmissionpolicystatus/controller.go

```diff
@@ -66,25 +84,52 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 	<-ctx.Done()
 }
 
-func NewController(policyInformer informerv1alpha1.ValidatingAdmissionPolicyInformer, policyClient admissionregistrationv1alpha1.ValidatingAdmissionPolicyInterface, typeChecker *validatingadmissionpolicy.TypeChecker) (*Controller, error) {
+func NewController(policyInformer informerv1alpha1.ValidatingAdmissionPolicyInformer, policyClient admissionregistrationv1alpha1.ValidatingAdmissionPolicyInterface, crdInformer informers.CustomResourceDefinitionInformer, crdClient apiextensionsv1client.CustomResourceDefinitionInterface, restMapper meta.RESTMapper) (*Controller, error) {
 	c := &Controller{
-		policyInformer: policyInformer,
-		policyQueue:    workqueue.NewRateLimitingQueueWithConfig(workqueue.DefaultControllerRateLimiter(), workqueue.RateLimitingQueueConfig{Name: ControllerName}),
-		policyClient:   policyClient,
-		typeChecker:    typeChecker,
+		policyInformer:           policyInformer,
+		policyQueue:              workqueue.NewRateLimitingQueueWithConfig(workqueue.DefaultControllerRateLimiter(), workqueue.RateLimitingQueueConfig{Name: ControllerName}),
+		policyClient:             policyClient,
+		crdInformer:              crdInformer,
+		crdClient:                crdClient,
+		restMapper:               restMapper,
+		compiledInSchemaResolver: resolver.NewDefinitionsSchemaResolver(scheme.Scheme, generated.GetOpenAPIDefinitions),
 	}
+	c.crdTracker = newCRDTracker(c)
 	reg, err := policyInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			c.enqueuePolicy(obj)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			c.enqueuePolicy(newObj)
 		},
+		DeleteFunc: func(obj interface{}) {
+			c.enqueuePolicy(obj)
```

Comment:
obj can be a DeletedFinalStateUnknown instance, which will panic enqueuePolicy

Reply:
`enqueuePolicy` does have the type assertion check so it should be good.

Reply:
oh, I see the check, so it won't panic, but it also won't correctly handle the policy... it needs to check for and unwrap a DeletedFinalStateUnknown object so we don't miscount references from deleted policies

Reply:
I've changed it to use `cache.DeletionHandlingMetaNamespaceKeyFunc` and `cache.MetaNamespaceKeyFunc`. These handle cluster-scoped resources well so I do not need to do it manually.